samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/evaluation/HistogramScoresBean.java

```diff
@@ -1237,4 +1246,176 @@ public String getExportStatisticsPdf() {
+  }
+
+  /**
+   * Get the Time stats for the variation as array based on bean.getAllSubmissions() getter:
+   *  - 1 for only best submissions (highest puntuation)
+   *  - 3 for all submissions type
+   * 
+   * @return timeStringArray (minimum time, average time and maximum time)
+   */
+  public ArrayList getTimeStatsVariationArray() {
+    ArrayList<Integer> scoreVariationArray = new ArrayList<>();
+    ArrayList timeStatsVariationArray = new ArrayList<>();
+    Double totalPossibleScoreDouble = 0.0;
+    try {
+      totalPossibleScoreDouble = Double.parseDouble(totalPossibleScore);
+    } catch (NumberFormatException e) {
+      totalPossibleScoreDouble = 0.0;
+    }
+    if (totalPossibleScoreDouble != 0) {
+      scoreVariationArray.add(50);
+      scoreVariationArray.add(70);
+      scoreVariationArray.add(90);
+
+      for (int minimumScore : scoreVariationArray) {
+        boolean firstTimeInside = true;
+        int timeElapsedInSeconds = 0;
+        int timeMin = 0;
+        int timeAvg = 0;
+        int timeMax = 0;
+        int counter = 0;
+        GradingService gradingService = new GradingService();
+        if (this.getAllSubmissions().equals("3")) {
+          TotalScoresBean totalBean = (TotalScoresBean) ContextUtil.lookupBean("totalScores");
+          List allSubmissionsList = gradingService.getAllSubmissions(this.getPublishedId());
+          for (Object submission : allSubmissionsList) {
+            AssessmentGradingData assessmentGradingAux = (AssessmentGradingData) submission;
+            if (assessmentGradingAux.getFinalScore() >= totalPossibleScoreDouble * minimumScore / 100){
+              try {
+                timeElapsedInSeconds = ((int) assessmentGradingAux.getSubmittedDate().getTime()) / 1000 - ((int) assessmentGradingAux.getAttemptDate().getTime()) / 1000;
+              } catch (Exception ex) {
+                timeElapsedInSeconds = 0;
+              }
+              timeAvg += timeElapsedInSeconds;
+              if (timeElapsedInSeconds > 0) {
+                counter++;
+              }
+            }
+            
+          }
+        } else {
+          TotalScoresBean totalBean = (TotalScoresBean) ContextUtil.lookupBean("totalScores");
+          Collection agents = totalBean.getAgents();
+          for (Object agent : agents) {
+            AgentResults agentResults = (AgentResults) agent;
+            if (agentResults.getAssessmentGradingId() != -1) {
+              AssessmentGradingData assessmentGradingAux = gradingService.load(agentResults.getAssessmentGradingId().toString());
+              if (assessmentGradingAux.getFinalScore() >= totalPossibleScoreDouble * minimumScore / 100){
+                try {
+                  timeElapsedInSeconds = ((int) agentResults.getSubmittedDate().getTime()) / 1000 - ((int) agentResults.getAttemptDate().getTime()) / 1000;
+                } catch (Exception ex) {
+                  timeElapsedInSeconds = 0;
```

Comment:
Add

log.error("Cannot resolve the submittedDate or the attemptDate, so it's a unanswered question, setting time as 0");

like on https://github.com/sakaiproject/sakai/pull/12334/files#diff-6efbfc756f2a2c4ad26c1e1fc5fdc24c2114df9618f98edf74dba074eaefa736R1308?